src/python/pants/core/util_rules/environments.py

```diff
@@ -327,89 +327,69 @@ class EnvironmentSensitiveOptionFieldMixin(ABC):
     option_name: str
 
 
-_rules_for_subsystems: dict[type[Subsystem], set[UnionRule]] = {}
+# Maps between non-list option value types and corresponding fields
+_SIMPLE_OPTIONS: dict[type, type[Field]] = {
+    str: StringField,
+}
 
+# Maps between the member types for list options. Each element is the
+# field type, and the `value` type for the field.
+_LIST_OPTIONS: dict[type, tuple[type[Field], type]] = {
+    str: (StringSequenceField, tuple[str, ...]),
```

Comment:
Py37 won't like the second member of the tuple because it is not a type annotation, it's runtime code. Have to use `typing.Tuple`

Reply:
Trying to use `typing.Tuple` caused `mypy` to tie itself in knots. Happily, as far as I can tell, the annotation on `value` is only ever used for static type checking, and all of the type checking is done based, so I was able to get rid of that bit entirely. Simpler again!